tests/test_data_cleaner.py

```diff
@@ -96,3 +99,114 @@ def current_size():
             self.assertTrue(removed == [])
 
             shutil.rmtree(temp_folder)
+
+    def test_early_termination(self):
+        temp_folder = tempfile.mkdtemp(suffix='_test_4')
+        sub_folder_1 = tempfile.mkdtemp(dir=temp_folder)
+        sub_folder_2 = tempfile.mkdtemp(dir=temp_folder)
+        sub_folder_3 = tempfile.mkdtemp(dir=temp_folder)
+
+        current_time = time.time()
+
+        def current_size(folder):
+            try:
+                folder_size = {temp_folder: 1024*2,
+                               sub_folder_2: 200,
+                               sub_folder_3: 1400}
+                return folder_size[folder]
+            except:
+                raise ValueError("Early termination failed")
+
+        def creation_time(folder):
+            folder_time = {sub_folder_1: int(current_time - 10),
+                           sub_folder_2: int(current_time - 20),
+                           sub_folder_3: int(current_time - 30)}
+            return folder_time[folder]
+
+        with mock.patch.object(DataCleaner, 'current_size') as mocked_current_size, \
+             mock.patch.object(DataCleaner, 'creation_time') as mocked_creation_time:
+                mocked_current_size.side_effect = current_size
+                mocked_creation_time.side_effect = creation_time
+
+                # remove folders older than 1 day
+                # and remove old folders as long as total size exceed 1 MB
+                mock_cleaner = DataCleaner(temp_folder, 1024, 60 * 60 * 24)
+                try:
+                    removed = mock_cleaner.clean_up(dry_run=False)
+                except:
+                    self.fail("Early termination failed")
+                self.assertTrue(len(removed) == 1)
+                self.assertEqual(removed[0], sub_folder_3)
+                shutil.rmtree(temp_folder)
+
+    def test_filterlist(self):
+        temp_folder = tempfile.mkdtemp(suffix='_test_5')
+        _, tmp_file_1 = tempfile.mkstemp(prefix='Person1', suffix='.dcm', dir=temp_folder)
+        _, tmp_file_2 = tempfile.mkstemp(prefix='Person2', suffix='.dcm', dir=temp_folder)
+
+        filter_list = ['Person1*.dcm']
+
+        with self.assertRaises(ValueError):
```

Comment:
Super cool that you do so many tests and they are pretty exhaustive. How about dividing this test into three tests - in general it's better to test only one thing, such that if some test fails we know immediately in which case it is.